test/rhsmlib/dbus/test_attach.py

```diff
@@ -10,187 +10,152 @@
-        self.interface = dbus.Interface(self.proxy, constants.ATTACH_INTERFACE)
-
-        attach_patcher = mock.patch("rhsmlib.dbus.objects.attach.AttachService", autospec=True)
-        self.mock_attach = attach_patcher.start().return_value
-        self.addCleanup(attach_patcher.stop)
-
-        entcertlib_patcher = mock.patch("rhsmlib.dbus.objects.attach.entcertlib.EntCertActionInvoker")
-        self.mock_action_invoker = entcertlib_patcher.start().return_value
-        self.addCleanup(entcertlib_patcher.stop)
-
-        self.mock_identity.is_valid.return_value = True
-        self.mock_identity.uuid = "id"
-
-        is_simple_content_access_patcher = mock.patch("rhsmlib.dbus.objects.attach.is_simple_content_access")
-        self.mock_is_simple_content_access = is_simple_content_access_patcher.start()
-        self.mock_is_simple_content_access.return_value = False
-        self.addCleanup(is_simple_content_access_patcher.stop)
-
-    def injection_definitions(self, *args, **kwargs):
-        if args[0] == inj.IDENTITY:
-            return self.mock_identity
-        elif args[0] == inj.CP_PROVIDER:
-            provider = mock.Mock(spec=CPProvider, name="CPProvider").return_value
-            provider.get_consumer_auth_cp.return_value = mock.Mock(name="MockCP")
-            return provider
-        else:
-            return None
-
-    def dbus_objects(self):
-        return [AttachDBusObject]
-
-    def test_pool_attach(self):
-        def assertions(*args):
-            expected_content = [json.dumps(CONTENT_JSON), json.dumps(CONTENT_JSON)]
-            result = args[0]
-            self.assertEqual(result, expected_content)
 
-        self.mock_attach.attach_pool.return_value = CONTENT_JSON
 
-        dbus_method_args = [["x", "y"], 1, {}, ""]
-        self.dbus_request(assertions, self.interface.PoolAttach, dbus_method_args)
+class TestAttachDBusObject(DBusServerStubProvider):
+    dbus_class = AttachDBusObject
+    dbus_class_kwargs: Dict[str, Any] = {}
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        is_simple_content_access_patch = mock.patch(
+            "rhsmlib.dbus.objects.attach.is_simple_content_access",
+            name="is_simple_content_access",
+        )
+        cls.patches["is_simple_content_access"] = is_simple_content_access_patch.start()
+        cls.addClassCleanup(is_simple_content_access_patch.stop)
+
+        is_registered_patch = mock.patch(
+            "rhsmlib.dbus.base_object.BaseObject.is_registered",
+            name="is_registered",
+        )
+        cls.patches["is_registered"] = is_registered_patch.start()
+        cls.addClassCleanup(is_registered_patch.stop)
+
+        update_patch = mock.patch(
+            "subscription_manager.certlib.BaseActionInvoker.update",
+            name="update",
+        )
+        cls.patches["update"] = update_patch.start()
+        cls.addClassCleanup(update_patch.stop)
+
+        super().setUpClass()
+
+    def setUp(self) -> None:
```

Comment:
this needs to call `setUp()` of the parent class (`DBusServerStubProvider`), so in case we add one implementation in the future it will be used as expected